lib/matplotlib/figure.py

```diff
@@ -2084,6 +2084,216 @@ def tight_layout(self, renderer=None, pad=1.08, h_pad=None, w_pad=None,
+
+            fig, axs = plt.subplots(2, 1)
+            axs[0].plot(np.arange(0, 1000, 50))
+            axs[0].set_ylabel('YLabel 0')
+            axs[1].set_ylabel('YLabel 1')
+            fig.align_ylabels()
+
+        """
+
+        from .tight_layout import get_renderer
+
+        if renderer is None:
+            renderer = get_renderer(self)
+
+        if axs is None:
+            axs = self.axes
+
+        axs = np.asarray(np.array(axs)).flatten().tolist()
+        for ax in axs:
+            _log.debug(' Working on: %s', ax.get_ylabel())
+            ss = ax.get_subplotspec()
+            nrows, ncols, row0, row1, col0, col1 = ss.get_rows_columns()
+            same = [ax]
+            labpo = ax.yaxis.get_label_position()
+            for axc in axs:
+                if axc != ax:
+                    if axc.yaxis.get_label_position() == labpo:
+                        ss = axc.get_subplotspec()
+                        nrows, ncols, row0, row1, colc0, colc1 = \
+                                ss.get_rows_columns()
+                        if (labpo == 'left') and (colc0 == col0):
+                            same += [axc]
+                        elif (labpo == 'right') and (colc1 == col1):
+                            same += [axc]
+            for axx in same:
+                _log.debug(' Same: %s', axx.yaxis.label)
+                axx.yaxis._align_label_siblings += [ax.yaxis]
+
+    # place holder until #9498 is merged...
+    def align_titles(self, axs=None, renderer=None):
+        """
+        Align the titles of subplots in this figure.
+
+        Parameters
+        ----------
+        axs : list of `~matplotlib.axes.Axes` (None)
+            Optional list of axes to align the xlabels.
+
+        renderer : (None)
+            Optional renderer to do the adjustment on.
+
+        See Also
+        --------
+        matplotlib.figure.Figure.align_xlabels
+
+        matplotlib.figure.Figure.align_ylabels
+        """
+
+        from .tight_layout import get_renderer
+
+        if renderer is None:
+            renderer = get_renderer(self)
+
+        if axs is None:
+            axs = self.axes
+
+        while len(axs):
+            ax = axs.pop()
+            ax._update_title_position(renderer)
+            same = [ax]
+            if ax._autolabelpos:
+                ss = ax.get_subplotspec()
+                nrows, ncols, row0, row1, col0, col1 = ss.get_rows_columns()
+                labpo = ax.xaxis.get_label_position()
+                for axc in axs:
+                    axc._update_title_position(renderer)
+                    if axc._autolabelpos:
+                        ss = axc.get_subplotspec()
+                        nrows, ncols, rowc0, rowc1, colc, col1 = \
+                                ss.get_rows_columns()
+                        if (rowc0 == row0):
+                            same += [axc]
+
+            x0, y0 = ax.title.get_position()
+            for axx in same:
```

Comment:
Looks easier (and cheaper) to get the max value of y in a first pass and then apply it throughout?